README.md

```diff
@@ -1,16 +1,439 @@
+
+## Getting Started
+
+Install the client SDK:
+
+```shell
+cargo add eventsourcingdb
+```
+
+Import the package and create an instance by providing the URL of your EventSourcingDB instance and the API token to use:
+
+```rust
+use eventsourcingdb::client::Client;
+
+// ...
+
+let base_url: Url = "localhost:3000".parse().unwrap();
+let api_token = "secret";
+let client = Client::new(base_url, api_token);
+```
+
+Then call the `ping` function to check whether the instance is reachable. If it is not, the function will return an error:
+
+```rust
+let result = client.ping().await;
+if let Err(err) = result {
+  // handle error...
+}
+```
+
+*Note that `Ping` does not require authentication, so the call may succeed even if the API token is invalid.*
+
+If you want to verify the API token, call `verify_api_token`. If the token is invalid, the function will return an error:
+
+```rust
+let result = client.verify_api_token().await;
+if let Err(err) = result {
+  // handle error...
+}
+```
+
+### Writing Events
+
+Call the `write_events` function and hand over a vector with one or more events. You do not have to provide all event fields – some are automatically added by the server.
+
+Specify `source`, `subject`, `type` (using `ty`), and `data` according to the [CloudEvents](https://docs.eventsourcingdb.io/fundamentals/cloud-events/) format.
+
+For `data` provide a json object using a `serde_json:Value`.
+
+The function returns the written events, including the fields added by the server:
+
+```rust
+let event = EventCandidate::builder()
+  .source("https://library.eventsourcingdb.io".to_string())
+  .subject("/books/42".to_string())
+  .ty("io.eventsourcingdb.library.book-acquired")
+  .data(json!({
+    "title": "2001 - A Space Odyssey",
+    "author": "Arthur C. Clarke",
+    "isbn": "978-0756906788",
+  }))
+  .build();
+
+let result = client.write_events(vec![event.clone()], vec![]).await;
+match result {
+  Ok(written_events) => // ...
+  Err(err) => // ...
+}
+```
+
+#### Using the `IsSubjectPristine` precondition
+
+If you only want to write events in case a subject (such as `/books/42`) does not yet have any events, use the `IsSubjectPristine` Precondition to create a precondition and pass it in a vector as the second argument:
+
+```rust
+let result = client.write_events(
+  vec![event.clone()],
+  vec![Precondition::IsSubjectPristine {
+    subject: "/books/42".to_string(),
+  }],
+).await;
+match result {
+  Ok(written_events) => // ...
+  Err(err) => // ...
+}
+```
+
+#### Using the `IsSubjectOnEventId` precondition
+
+If you only want to write events in case the last event of a subject (such as `/books/42`) has a specific ID (e.g., `0`), use the `IsSubjectOnEventID` Precondition to create a precondition and pass it in a vector as the second argument:
```

Comment:
```suggestion
If you only want to write events in case the last event of a subject (such as `/books/42`) has a specific ID (e.g., `0`), use the `IsSubjectOnEventID` precondition to create a precondition and pass it in a vector as the second argument:
```